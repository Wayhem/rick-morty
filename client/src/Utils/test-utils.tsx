import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult, Queries } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Provider } from 'react-redux'
import { MemoryRouter as Router } from 'react-router-dom'
import { store } from 'Store'
import { theme } from 'Config/styled'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AllTheProviders: FC = ({ children }: { children?: any }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  </ThemeProvider>
)

AllTheProviders.defaultProps = {
  children: <></>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult<Queries, HTMLElement> => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'

export { customRender as render }
