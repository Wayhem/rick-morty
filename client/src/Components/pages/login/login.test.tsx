import React from 'react'
import Login from 'Components/pages/login'
import { render } from 'Utils/test-utils'

test('Login renders and matches snapshot', () => {
  const { container } = render(<Login />)
  expect(container).toMatchSnapshot()
})
