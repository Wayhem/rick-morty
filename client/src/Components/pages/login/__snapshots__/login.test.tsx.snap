// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders and matches snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ gslMLe"
  >
    <div
      class="sc-jSgupP iDAwLf"
    >
      <form
        autocomplete="off"
        class="sc-jrAGrp bQeBPC"
      >
        <div
          class="sc-iqHYGH glFaZH"
        >
          <div
            class="sc-gKsewC jXmpdJ"
          >
            <div
              class="sc-gsTCUz khJLZs"
            >
              <input
                class="sc-dlfnbm tELdO"
                id="emailInput"
                placeholder="email"
                required=""
                type="email"
                value=""
              />
              <label
                for="emailInput"
              >
                Email
              </label>
            </div>
          </div>
          <div
            class="sc-gKsewC jXmpdJ"
          >
            <div
              class="sc-gsTCUz khJLZs"
            >
              <input
                class="sc-dlfnbm kReXYC"
                id="usernameInput"
                placeholder="username"
                required=""
                type="text"
                value=""
              />
              <label
                for="usernameInput"
              >
                Username
              </label>
            </div>
          </div>
          <div
            class="sc-gKsewC jXmpdJ"
          >
            <div
              class="sc-gsTCUz khJLZs"
            >
              <input
                class="sc-dlfnbm kReXYC"
                id="passwordInput"
                placeholder="password"
                required=""
                type="password"
                value=""
              />
              <label
                for="passwordInput"
              >
                Password
              </label>
            </div>
          </div>
        </div>
        <div
          class="sc-pFZIQ fmHHoo"
        >
          <button
            class="sc-hKgILt sc-fubCfw geIEcq"
            type="button"
          >
            Clear
          </button>
          <button
            class="sc-eCssSg sc-iBPRYJ hTuqml"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
      <div
        class="sc-pFZIQ hJgdOq"
      >
        <div
          class="sc-kEjbxe hEgTGC"
        >
          Register
        </div>
        <div
          class="sc-kEjbxe ksQVmt"
        >
          Login
        </div>
      </div>
    </div>
  </div>
</div>
`;
